pints/_mcmc/_mala.py

```diff
@@ -0,0 +1,319 @@
+        """
+        if epsilon is None:
+            self._epsilon = self._step_size * self._scale_vector
+        else:
+            a = np.atleast_1d(epsilon)
+            if not len(a) == self._n_parameters:
+                raise ValueError('Dimensions of epsilon must be same as ' +
+                                 'number of parameters')
+            for element in epsilon:
+                if element <= 0:
+                    raise ValueError('Elements of epsilon must exceed 0')
+            self._epsilon = np.array(epsilon)
+
+    def epsilon(self):
+        """
+        Returns epsilon which is the effective step size used in proposals
+        """
+        return self._epsilon
+
+    def scale_vector(self):
+        """
+        Returns vector used to scale step size for proposals
+        """
+        return self._scale_vector
+
+    def step_size(self):
+        """
+        Returns step size used to propose new points
+        """
+        return self._step_size
+
+    def acceptance_rate(self):
+        """
+        Returns the current (measured) acceptance rate
+        """
+        return self._acceptance
+
+    def ask(self):
+        """ See :meth:`SingleChainMCMC.ask()`. """
+        # Initialise on first call
+        if not self._running:
+            self._initialise()
+
+        if self._ready_for_tell:
+            raise RuntimeError('Ask() called when expecting call to tell().')
+
+        # Propose new point
+        if self._proposed is None:
+            self._forward_mu = self._current + (self._epsilon**2 / 2.0) * (
+                self._current_gradient)
+            self._proposed = np.random.multivariate_normal(
+                self._forward_mu,
+                self._epsilon**2 * np.diag(np.ones(self._n_parameters)))
+
+            self._forward_q = scipy.stats.multivariate_normal.logpdf(
+                self._proposed, self._forward_mu,
+                self._epsilon**2 * np.diag(np.ones(self._n_parameters))
+            )
+
+            # Set as read-only
+            self._proposed.setflags(write=False)
+
+        self._ready_for_tell = True
+        # Return proposed point
+        return self._proposed
+
+    def tell(self, reply):
+        """ See :meth:`pints.SingleChainMCMC.tell()`. """
+        # Check if we had a proposal
+        if self._proposed is None:
+            raise RuntimeError('Tell called before proposal was set.')
+        if not self._ready_for_tell:
+            raise RuntimeError('Tell called before proposal was set.')
+        self._ready_for_tell = False
+
+        # Ensure fx is a float
+        fx, log_gradient = reply
+
+        # First point?
+        if self._current is None:
+            if not np.isfinite(fx):
+                raise ValueError(
+                    'Initial point for MCMC must have finite log_pdf.')
+
+            # Accept
+            self._current = self._proposed
+            self._current_log_pdf = fx
+            self._current_gradient = log_gradient
```

Comment:
Think this might need a minus sign, to match with what happens with the proposed_gradient below?

Reply:
@MichaelClerx Good spot, I think the old sign was wrong below (which I have now corrected)